design/overview.md

```diff
@@ -104,6 +106,17 @@ At this point, we know we have a contract on Ethereum with the proper *MultiSig
 
 Note: X% is some security factor for bootstrapping. 67% is sufficient to release, but we don't want to start until there is a margin of error online (not to fall off with a small change of voting power). This may be 70, 80, 90, or even 95% depending on how much assurances we want that all *Orchestrators* are operational before starting.
 
+## Observing Ethereum state ("Ethereum Oracle")
```

Comment:
You mean that our oracle concept is general right? becuase I don't yet agree that we should generalize the oracle. I would rather boilerplate over trying to do that at the moment. It's a rabbit hole.

Reply:
I would rather generalize from the bottom up (combine backend functions to minimize biolerplate but keep separate oracle message types) than top down.

Reply:
this seems to be what @alpe  is already working on https://github.com/alpe/peggy/pull/1

Reply:
Ultimately, I'm not the one working on the Cosmos code right now, but all you really need is one message type to pull in all the information you need from Eth for every block. From there it can just be made available in a store. Checking Geth and sending a separate message for arbitrary different things that may be needed from eth seems like a lot of extra work, and harder to understand with no benefit. But not my call i guess.

Reply:
There is some potential to generalize the oracle claims but this is an optimization we can do later. Let's see how much mapping is required in the orchestrator.